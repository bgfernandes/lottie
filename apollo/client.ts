
import { ApolloClient, InMemoryCache, NormalizedCacheObject } from '@apollo/client'

let apolloClient: ApolloClient<NormalizedCacheObject>

if (typeof window === 'undefined') {
  // code is running in the server side
  const { SchemaLink } = require('@apollo/client/link/schema')
  const { schema } = require('./schema')

  apolloClient = new ApolloClient({
    ssrMode: true,
    link: new SchemaLink({ schema }),
    cache: new InMemoryCache(),
  })
} else {
  // code is running in the browser
  const { HttpLink } = require('@apollo/client/link/http')

  apolloClient = new ApolloClient({
    ssrMode: false,
    link: new HttpLink({
      uri: '/api/graphql',
      credentials: 'same-origin',
    }),
    cache: new InMemoryCache(),
  })
}

export default apolloClient
