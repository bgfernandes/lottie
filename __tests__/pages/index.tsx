/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { MockedProvider } from '@apollo/client/testing'
import Home, { HELLO_MESSAGE_QUERY } from '../../pages/index'

const mockServerSideData = {
  hello: { message: 'Hello from Apollo.' }
}

const mockClientSideData = [
  {
    request: {
      query: HELLO_MESSAGE_QUERY
    },
    result: {
      data: {
        hello: { message: 'Hello from Apollo.' }
      }
    }
  }
]

describe('Home', () => {
  it('renders hello world', () => {
    const { getByText } = render(
      <MockedProvider mocks={mockClientSideData} addTypename={false}>
        <Home serverSideData={mockServerSideData} />
      </MockedProvider>
    )

    const helloWorld = getByText('Hello world.')

    expect(helloWorld).toBeInTheDocument()
  })

  it('matches the snapshot', () => {
    const tree = renderer.create(
      <MockedProvider mocks={mockClientSideData} addTypename={false}>
        <Home serverSideData={mockServerSideData} />
      </MockedProvider>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('shows the client side fetched message', async () => {
    const { getByText } = render(
      <MockedProvider mocks={mockClientSideData} addTypename={false}>
        <Home serverSideData={mockServerSideData} />
      </MockedProvider>
    )

    await new Promise(resolve => setTimeout(resolve, 0))

    const clientSideMessage = getByText('This message was fetched from the graphql API on the client side: Hello from Apollo.')

    expect(clientSideMessage).toBeInTheDocument()
  })
})
