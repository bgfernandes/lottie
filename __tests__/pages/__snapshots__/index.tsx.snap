// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div>
  Hello world.
  <p
    className="flex justify-center"
  >
    This page is using tailwindcss.
  </p>
  <p>
    This message was fetched from the graphql API on the server side: 
    Hello from Apollo.
  </p>
  <p>
    This message was fetched from the graphql API on the client side: 
    loading
  </p>
</div>
`;
